dali/base/dadfs.cpp

```diff
@@ -5882,6 +5909,32 @@ class CDistributedSuperFile: public CDistributedFileBase<IDistributedSuperFile>
             throw MakeStringException(-1,"addSubFile: File %s is already a subfile of %s", sub->queryLogicalName(),queryLogicalName());
     }
 
+    void validate()
+    {
+        IDistributedFileDirectory &dfd = queryDistributedFileDirectory();
+        unsigned numSubfiles = root->getPropInt("@numsubfiles",0);
+        if (numSubfiles)
+        {
+            Owned<IPropertyTreeIterator> treeIter = root->getElements("SubFile");
+            unsigned subFileCount = 0;
+            ForEach(*treeIter)
+            {
+                IPropertyTree & st = treeIter->query();
+                StringBuffer subfilename;
+                st.getProp("@name", subfilename);
+                if (!dfd.exists(subfilename.str(), NULL))
+                    throw MakeStringException(-1, "Logical subfile '%s' doesn't exists!", subfilename.str());
+
+                if (!dfd.isSuperFile(subfilename.str(), NULL))
+                    if (!dfd.existsPhysical(subfilename.str(), NULL))
+                        throw MakeStringException(-1, "Physical Parts doesn't exists for logical file %s!", subfilename.str());
+                subFileCount++;
+            }
+            if (numSubfiles != subFileCount)
+                throw MakeStringException(-1, "The value of @numsubfiles (%d) not equal with the number of SubFile items (%d)!",numSubfiles, subFileCount);
```

Comment:
trivial: "The value of @numsubfiles (%d) is not equal to the number of SubFile items (%d)!"